src/ucs/sys/sys.c

```diff
@@ -541,7 +541,11 @@ ucs_status_t ucs_sysv_alloc(size_t *size, void **address_p, int flags, int *shmi
     }
 
     /* Attach segment */
-    ptr = shmat(*shmid, NULL, 0);
+    if (*address_p) {
```

Comment:
minor: can we shorten to
`ptr = shmat(*shmid, *address_p, (*address_p) ? SHM_REMAP : 0);`